src/main.rs

```diff
@@ -16,6 +16,11 @@ use crate::network::NetworkIterator;
 use crate::vendor::Vendor;
 
 fn main() {
+    // Upgrade user privileges when needed
+    // ----------------------------------------
+    // Providing a prompt for the user when
+    // the app is run and user is not root    
+    sudo::escalate_if_needed().expect("You need root permissions to run this app. Unable to escalate to sudo");
```

Comment:
Great @gsuyemoto ! A small change : the `escalate_if_needed()` call can be moved after the `scan_options.request_protocol_print()` condition (L48) in order to allow non-root users to view the protocol ASCII print and list the interfaces.

Reply:
Sure! I'll move the escalate, makes total sense!